src/common/helper/blocks.cpp

```diff
@@ -96,79 +97,110 @@ uint32_t get_users_count(uint16_t bank) {
     return (st.st_size/sizeof(user_t));
 }
 
-void db_backup(uint32_t block_path, uint16_t nodes) {
+void db_backup(uint32_t block_path, uint16_t nodes)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this function to reduce its Cognitive Complexity from 39 to the 25 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3776)